### PR TITLE
PT-61 PT-45 [Ethereum] - Remove legacy MetaProperties endpoint and DT…

### PR DESCRIPTION
**Pull request by tgenman**
…O from openapi

### CODE REVIEW

File: nft/api/src/main/kotlin/com/rarible/protocol/nft/api/controller/ItemController.kt [hunk 1]
```diff
@@ -62,62 +54,29 @@ class ItemController(
...
+            .getById(conversionService.convert(itemId))
+            .let { conversionService.convert<NftItemDto>(it) }
         return ResponseEntity.ok(result)
     }
 
     override suspend fun getNftLazyItemById(itemId: String): ResponseEntity<LazyNftDto> {
-        val result = itemService.getLazy(conversionService.convert(itemId))
+        val result = itemService
+            .getLazyById(conversionService.convert(itemId))
+            .let { conversionService.convert<LazyNftDto>(it) }
```
Comment (zolt) [inline nft/api/src/main/kotlin/com/rarible/protocol/nft/api/controller/ItemController.kt:67]: if I understood correctly, `getLazyById` have non-null return type, why we use `let { ... }` here?

File: nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt [hunk 2]
```diff
@@ -2,72 +2,49 @@ package com.rarible.protocol.nft.api.service.item
...
-                itemMetaService.getAvailableMeta(itemId, demander)
-            }
-            val extendedItem = ExtendedItem(item.await(), itemMeta.await())
-            conversionService.convert(extendedItem)
-        }
-    }
+    suspend fun getById(itemId: ItemId): Item =
+        itemRepository
+            .findById(itemId)
+            .awaitSingleOrNull() ?: throw EntityNotFoundApiException("Item", itemId)
```
Comment (zolt) [inline nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt:42]: `awaitSingleOrNull()` is deprecated
can be replacedBy `awaitFirstOrElse {  throw EntityNotFoundApiException("Item", itemId) }`

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/model/ContentMeta.kt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package com.rarible.protocol.nft.core.model
 
 @Deprecated("Ethereum API will soon be not responsible for loading content meta. ")
-data class ContentMeta(
+data class ContentMeta(  // TODO Remove
```
Comment (zolt) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/model/ContentMeta.kt:4]: why not remove?

Comment (tgenman) [reply nft/core/src/main/kotlin/com/rarible/protocol/nft/core/model/ContentMeta.kt:4]: Will be removed in 3th part of this task.
PR is not finished yet

File: nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt (hunk 2 above)
Comment (tgenman) [reply nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt:42]: Sorry, cannot find info about deprecation
Not in Idea, not in [doc](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-reactor/kotlinx.coroutines.reactor/await-single-or-null.html)

**Review by zolt — APPROVED** [review]

File: nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt [hunk 4]
```diff
@@ -81,58 +58,33 @@ class ItemService(
...
-        metaLoadingDemander: String
-    ): List<ExtendedItem> = coroutineScope {
-        val items = itemRepository.searchByIds(list)
-        items.mapAsync { item ->
-            val meta = getAvailableMetaWithoutScheduling(item.id)
-            ExtendedItem(item, meta)
-        }
+        itemRepository.searchByIds(ownerships.keys)
+            .mapAsync { item ->
+                item.copy(
```
Comment (yakischik) [inline nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt:78]: async not needed here, i think

File: nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt [hunk 5]
```diff
@@ -81,58 +58,33 @@ class ItemService(
...
-    private suspend fun getAvailableMetaWithoutScheduling(itemId: ItemId): ItemMeta? {
-        return itemMetaService.getAvailableMetaOrLoadSynchronously(
-            itemId = itemId,
-            demander = "search",
-            synchronous = false,
-            useMetaCache = true,
-            scheduleIfNeeded = false
-        )
-    }
+    suspend fun search(ids: Set<ItemId>): List<Item> = itemRepository.searchByIds(ids)
```
Comment (yakischik) [inline nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt:87]: Lets call such function getAll, search is something required complex filtering/conditions

File: nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/mint/LegacyMintService.kt [hunk 6]
```diff
@@ -48,7 +46,6 @@ class LegacyMintService(
                 date = nowMillis()
             )
         ).awaitFirst()
-        itemMetaService.removeMeta(itemId, "burn lazy mint")
```
Comment (yakischik) [inline nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/mint/LegacyMintService.kt:51]: Check

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/converters/dto/NftItemMetaDtoConverter.kt [hunk 7]
```diff
@@ -33,84 +25,18 @@ class NftItemMetaDtoConverter(
...
-            animation = createAnimationMedia(source, itemIdDecimalValue),
             createdAt = source.properties.createdAt,
             tags = source.properties.tags,
             genres = source.properties.genres,
             language = source.properties.language,
             rights = source.properties.rights,
             rightsUri = source.properties.rightsUri,
             externalUri = source.properties.externalUri,
-            content = source.properties.content.map { EthMetaContentConverter.convert(it) },
+            content = source.properties.content.asList().map { EthMetaContentConverter.convert(it) },
```
Comment (yakischik) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/converters/dto/NftItemMetaDtoConverter.kt:35]: Why asList?

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/ItemMetaService.kt [hunk 8]
```diff
@@ -17,89 +13,37 @@ import java.time.Duration
...
 ) {
-    private val logger = LoggerFactory.getLogger(ItemMetaService::class.java)
-
-    /**
-     * Return available meta or `null` if it hasn't been loaded,
-     * has failed, or hasn't been requested yet.
-     * Schedule an update in the last case.
-     */
-    suspend fun getAvailableMetaOrScheduleLoading(
+    suspend fun getAvailableMeta(
```
Comment (yakischik) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/ItemMetaService.kt:18]: 'Available' word is not actual anymore

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/ItemMetaService.kt [hunk 9]
```diff
@@ -17,89 +13,37 @@ import java.time.Duration
...
-            itemId = itemId,
-            useMetaCache = true,
-            synchronous = false,
-            demander = demander
-        )
+        return itemMeta
     }
 
-    suspend fun getAvailableMetaOrLoadSynchronouslyWithTimeout(
+    suspend fun getAvailableMetaWithTimeout(
```
Comment (yakischik) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/ItemMetaService.kt:38]: 'Available' word is not actual anymore

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/ItemMetaService.kt [hunk 10]
```diff
@@ -112,33 +56,18 @@ class ItemMetaService(
...
         val itemMeta = itemMetaResolver.resolvePendingItemMeta(itemId, tokenUri) ?: return
-        itemMetaCacheLoaderService.save(itemId.toCacheKey(), itemMeta)
+        // TODO in PT-566
         logMetaLoading(itemId, "resolved and saved meta for a pending item by $tokenUri: $itemMeta")
     }
-}
 
-fun ItemId.toCacheKey(): String = decimalStringValue
+    companion object {
+        private val logger = LoggerFactory.getLogger(ItemMetaService::class.java)
```
Comment (yakischik) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/ItemMetaService.kt:71]: There is no sense to do logger static in beans

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/properties/ContentBuilder.kt [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+object ContentBuilder {
+
+    fun getItemMetaContent(
+        imageOriginal: String? = null,
+        imageBig: String? = null,
+        imagePreview: String? = null,
+        videoOriginal: String? = null
+    ): ItemMetaContent {
+        return ItemMetaContent(
+            imageOriginal = imageOriginal?.let { toImage(it, MetaContentDto.Representation.ORIGINAL) },
```
Comment (yakischik) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/properties/ContentBuilder.kt:21]: Check not only null, but isBlank() too

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/token/meta/TokenMetaService.kt [hunk 12]
```diff
@@ -21,43 +15,33 @@ import scalether.domain.Address
...
 
-        // TODO originally, it should be done in resolve procedure
-        val withContent = properties?.image?.let {
-            properties.copy(
-                content = listOf(
-                    EthMetaContent(it, MetaContentDto.Representation.ORIGINAL, null, EthImageProperties())
-                )
-            )
-        } ?: properties
+        if (properties == null || properties.content.imageOriginal?.url == null) return TokenMeta.EMPTY
```
Comment (yakischik) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/token/meta/TokenMetaService.kt:24]: isBlank()

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/properties/ContentBuilder.kt [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+            imageBig = imageBig?.let { toImage(it, MetaContentDto.Representation.BIG) },
+            imagePreview = imagePreview?.let { toImage(it, MetaContentDto.Representation.PREVIEW) },
+            videoOriginal = videoOriginal?.let { toVideo(it, MetaContentDto.Representation.ORIGINAL) }
```
Comment (yakischik) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/properties/ContentBuilder.kt:24]: Also we need to trim leading/trailing spaces from URL

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/token/meta/TokenMetaService.kt [hunk 14]
```diff
@@ -21,43 +15,33 @@ import scalether.domain.Address
...
-                properties = withContent,
-                contentMeta = withContent.image?.let {
-                    mediaMetaService.getMediaMetaFromCache(it, id.prefixed())
-                }
+        mediaMetaService.getMediaMetaFromCache(imageOriginal.url, id.prefixed())
+        return TokenMeta(
+            properties = properties.copy(
+                content = TokenMetaContent(
+                    ContentBuilder.populateContent(
+                        ethMetaContent = imageOriginal,
```
Comment (yakischik) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/token/meta/TokenMetaService.kt:33]: Trim leading/trailing spaces

**Pull request merged**